cli/azd/pkg/tools/git/git.go

```diff
@@ -143,6 +143,13 @@ func (cli *gitCli) InitRepo(ctx context.Context, repositoryPath string) error {
 		return fmt.Errorf("failed to init repository: %s: %w", res.String(), err)
 	}
 
+	// Set initial branch to main
+	runArgs = exec.NewRunArgs("git", "-C", repositoryPath, "checkout", "-b", "main")
```

Comment:
```suggestion
	runArgs = exec.NewRunArgs("git", "-C", repositoryPath, "branch", "-M", "main")
```

nit: Consider using branch -M instead of `checkout -b main` With branch -M, it renames the existing branch, with checkout -b it persists the old branch. For newly inited repos it doesn't matter, but if a user manually ran `git init` with an older version of git, both master and main will appear in the branch list.

In addition, github points users at `branch -M main` for newly created repos.

https://git-scm.com/docs/git-branch#Documentation/git-branch.txt--M

Reply:
This did not work
I created a new PR to fix it: https://github.com/Azure/azure-dev/pull/777

The initial branch does not really exists until the first ever commit is done